### PR TITLE
cleanup(test/libscap): ensure each libscap test suite is activated, deprecate old userspace/libscap/test

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

/area libscap

> /area libpman

> /area libsinsp

/area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

- `generic` subfolder caused the respective test suites not to be activated -> fixed by unifying `generic` and `tables`  under `userspace` (open to different naming, still confused why there were issues before ...)
- ported missing common_strl unit test to test/libscap
- fixed one broken scap_ppm_sc test now that they are activated
- deprecate old `userspace/libscap/test` unit test suite in favor of `test/libscap`

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/libscap/test_suites/userspace/scap_ppm_sc.cpp [hunk 1]
```diff
@@ -100,8 +100,8 @@ TEST(scap_ppm_sc, scap_get_events_from_ppm_sc)
 			struct syscall_evt_pair pair = g_syscall_table[sys_id];
 			if(pair.ppm_sc == ppm_sc)
 			{
-				ASSERT_TRUE(events_array[pair.enter_event_type]) << "ppm_sc: " << scap_get_syscall_info_table()[pair.ppm_sc].name << " (" << pair.ppm_sc << ") should be associated with event: " << pair.enter_event_type << std::endl;
-				ASSERT_TRUE(events_array[pair.exit_event_type]) << "ppm_sc: " << scap_get_syscall_info_table()[pair.ppm_sc].name << " (" << pair.ppm_sc << ") should be associated with event: " << pair.exit_event_type << std::endl;
+				ASSERT_TRUE(events_array[pair.enter_event_type]) << "ppm_sc: " << scap_get_ppm_sc_name((ppm_sc_code)pair.ppm_sc) << " (" << pair.ppm_sc << ") should be associated with event: " << pair.enter_event_type << std::endl;
```
Comment (incertum) [inline test/libscap/test_suites/userspace/scap_ppm_sc.cpp:103]: @FedeDP and @Andreagit97 plz confirm, this was the broken test we didn't notice.

Btw still unsure why the `generic` subfolder just wasn't activated, not like it makes a lot of sense. I quickly just unified all of them into `userspace`, but open to refactor to something else while ensuring each test suite is loaded.

@Andreagit97 only test that was not ported over to test/libscap was the recently added common_strl, it's now ported over.

Comment (Andreagit97) [reply test/libscap/test_suites/userspace/scap_ppm_sc.cpp:103]: Yep for some reason they were skipped :/ Great catch!

**Review by Andreagit97 — COMMENTED** [review]
Thank you very much for this! i've just enabled by default the gvisor tests

**Review by Andreagit97 — APPROVED** [review]
/approve
Now all tests correctly run! thank you

**Comment by incertum** [comment]
> ... i've just enabled by default the gvisor tests

oh and I didn't catch that one, ty @Andreagit97!

**Pull request merged**